core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -543,6 +543,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    if (contains("spark.cores.max")) {
+      val totalCores = getInt("spark.cores.max", -1)
```

Comment:
@jerryshao I don't understand very cleanly, if we don't set this configuration ,the "if (contains("spark.cores.max")) " will not got into.